atc/engine/builder/builder.go

```diff
@@ -18,7 +17,8 @@ const supportedSchema = "exec.v2"
 type StepFactory interface {
 	GetStep(atc.Plan, exec.StepMetadata, db.ContainerMetadata, exec.GetDelegate) exec.Step
 	PutStep(atc.Plan, exec.StepMetadata, db.ContainerMetadata, exec.PutDelegate) exec.Step
-	TaskStep(atc.Plan, exec.StepMetadata, db.ContainerMetadata, exec.TaskDelegate, lock.LockFactory) exec.Step
```

Comment:
why are we removing the lockFactory from TaskStep? We need it for `limit-active-tasks`.

Reply:
We don't need to pass it in because the `stepFactory` already has a `lockFactory`.

https://github.com/concourse/concourse/pull/4202/files#diff-7e37c202e7044cc08233052392ecdb38R146

Reply:
ah ok, missed that, thanks 👍